lib/plug/conn.ex

```diff
@@ -1696,6 +1696,22 @@ defmodule Plug.Conn do
     conn |> get_session |> Map.get(session_key(key))
   end
 
+  @doc """
+  Returns session value for the given `key`.
+
+  Returns the default value if `key` does not exist or is `nil`.
```

Comment:
Typically get functions only returns the default value if `key` does not exist. Can you please change it accordingly?

Reply:
> Typically get functions only returns the default value if `key` does not exist. Can you please change it accordingly?

Sounds good!
I see two obvious paths here but I'm unsure which you'd see as more idiomatic.
1. Use `Map.has_key?` and perform `Map.get` when true.

```elixir
    session = get_session(conn)
    key = session_key(key)

    case Map.has_key?(session, key) do
      false -> default
      true -> Map.get(session, key)
    end
```

2. Use `Map.get` with a default value of `:missing_session_key` and switch on that.
```elixir
    session_value = conn |> get_session |> Map.get(session_key(key), :missing_session_value)

    case session_value do
      :missing_session_value -> default
      value -> value
    end
```

Unsure if removing a single function call is worth the potential obfuscation.

I've pushed a commit with option 1 for now!

Reply:
Actually I suppose
```elixir
    session_value = conn |> get_session |> Map.fetch(session_key(key))

    case session_value do
      {:ok, value} -> value
      :error -> default
    end
```
makes the most sense, pushing that instead!